### ISSUE TITLE
ProtocolManager using wrong network ID

### ISSUE THREAD

**Issue by JVillella** [Fantom-foundation/go-opera#281]
https://github.com/Fantom-foundation/go-opera/blob/10ca669279e26d21133c1d95db26ea41a6479360/gossip/handler.go#L80

`net` is not set, which leads to `NetworkID` initializing to 0 (should be 250). I logged the message being passed during devp2p handshake and can confirm this is the case.

```
Apr 04 16:36:41 opera[3923024]: INFO [04-04|16:36:41.096] HANDSHAKE                                 pm.net.NetworkID=0
```

Because ENR entry creation pulls it from service store it gets set correctly there,

```go
func currentENREntry(svc *Service) *enrEntry {
	genesisHash := *svc.store.GetGenesisHash()
	return &enrEntry{
		ForkID: forkid.NewID(svc.store.GetRules().EvmChainConfig(), common.Hash(genesisHash), uint64(svc.store.GetLatestBlockIndex())),
	}
}
```

The mismatch is... a little unpleasant :)

**Comment by quan8** [comment]
Hi @JVillella , which branch are you looking at? Please try release/1.1.0-rc.4.

**Comment by JVillella** [comment]
I am on release/1.0.2-rc.7

I guess it was updated recently. It looks like this got fixed in the 1.1.0-rc.4. I see there was a workaround for this incompatibility made here,

https://github.com/Fantom-foundation/go-opera/blob/8d3ee1a192057046a1eb3befbe985aa9c4d7147d/gossip/peer.go#L536-L538

Thanks!
